src/Uno.UWP/Graphics/Display/DisplayInformation.Android.cs

```diff
@@ -1,22 +1,184 @@
+			RawDpiY = realDisplayMetrics.Ydpi;
+			ScreenWidthInRawPixels = (uint)realDisplayMetrics.WidthPixels;
+			ScreenHeightInRawPixels = (uint)realDisplayMetrics.HeightPixels;
+			RawPixelsPerViewPixel = 1.0f * (int)realDisplayMetrics.DensityDpi / (int)DisplayMetricsDensity.Default;
+
+			var x = Math.Pow(ScreenWidthInRawPixels / realDisplayMetrics.Xdpi, 2);
+			var y = Math.Pow(ScreenHeightInRawPixels / realDisplayMetrics.Ydpi, 2);
+			var screenInches = Math.Sqrt(x + y);
+			DiagonalSizeInInches = screenInches;
+		}
+
+		/// <summary>
+		/// Sets the NativeOrientation property
+		/// </summary>
+		/// <param name="windowManager">Window manager</param>
+		/// <remarks>
+		/// Based on responses in
+		/// <see cref="https://stackoverflow.com/questions/4553650/how-to-check-device-natural-default-orientation-on-android-i-e-get-landscape">this SO question</see>
+		/// </remarks>
+		private void UpdateNativeOrientation(IWindowManager windowManager)
+		{
+			var orientation = ContextHelper.Current.Resources.Configuration.Orientation;
+			if (orientation == Android.Content.Res.Orientation.Undefined)
+			{
+				NativeOrientation = DisplayOrientations.None;
+				return;
+			}
+
+			var rotation = windowManager.DefaultDisplay.Rotation;
+			bool isLandscape;
+			switch (rotation)
+			{
+				case SurfaceOrientation.Rotation0:
+				case SurfaceOrientation.Rotation180:
+					isLandscape = orientation == Android.Content.Res.Orientation.Landscape;
+					break;
+				default:
+					isLandscape = orientation == Android.Content.Res.Orientation.Portrait;
+					break;
+			}
+			NativeOrientation = isLandscape ? DisplayOrientations.Landscape : DisplayOrientations.Portrait;
+		}
+
+		/// <summary>
+		/// Sets the CurrentOrientation property
+		/// </summary>
+		/// <param name="windowManager">Window manager</param>
+		/// <remarks>
+		/// Again quite complicated to do on Android, based on accepted solution at
+		/// <see cref="https://stackoverflow.com/questions/10380989/how-do-i-get-the-current-orientation-activityinfo-screen-orientation-of-an-a">this SO question</see>
+		/// </remarks>
+		private void UpdateCurrentOrientation(IWindowManager windowManager)
+		{
+			var rotation = windowManager.DefaultDisplay.Rotation;
+			var displayMetrics = new DisplayMetrics();
+			windowManager.DefaultDisplay.GetMetrics(displayMetrics);
+
+			int width = displayMetrics.WidthPixels;
+			int height = displayMetrics.HeightPixels;
+
+			if ( width == height)
+			{
+				//square device, can't tell orientation
+				CurrentOrientation = DisplayOrientations.None;
+				return;
+			}
+			
+			var portraitOrientation0Or180 =
+				(rotation == SurfaceOrientation.Rotation0 ||
+				rotation == SurfaceOrientation.Rotation180)
+				&& height > width;
+			var portraitOrientation90Or270 =
+				(rotation == SurfaceOrientation.Rotation90 ||
+				rotation == SurfaceOrientation.Rotation270)
+				&& width > height;
+
+			// if the device's natural orientation is portrait:
```

Comment:
Could you just use the NativeOrientation property here? Why is the logic slightly different?

Reply:
You are right, this could probably be simplified/unified, I will take a look

Reply:
Done :-)